### PR TITLE
Feature 9 skill system

### PR DESCRIPTION
**Pull request by charlietaoxl**
Created Skill class and Type enum in battle package. Implemented Skill class with setters, getters, and find functions.

### CODE REVIEW

**Review by Yasutakahi — CHANGES_REQUESTED** [review]
You might want to add some doc string but the code itself look very nice to me

**Review by marjaleenakiik — APPROVED** [review]
Code makes sense, and docstrings are included for more complex methods.

File: src/main/java/objects/battle/Skill.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+public class Skill {
+    /** A class for all skills to be used by any kinds of Characters in battles.
+     * Attributes:
+     * allSkills: A dictionary mapping the string name of a skill to the skill object itself
+     * name: A string representing the name of the skill
+     * damage: An integer representing how much damage the skill does
+     * lag; An integer that adds priority to skills
```
Comment (juhwansonn) [inline src/main/java/objects/battle/Skill.java:12]: You might want to fix a typo here.

**Review by juhwansonn — COMMENTED** [review]
There is a typo in Skill.java, other than that everything looks good.

**Review by juhwansonn — DISMISSED** [review]

**Review by marjaleenakiik — APPROVED** [review]
Skills look functional, docstrings describe them well.

**Review by Yasutakahi — APPROVED** [review]
doc string is added and looks very nice

**Review by bw55555 — APPROVED** [review]
Entity looks good and follows clean architecture

**Pull request merged**